### ISSUE TITLE
Any idea to get r2 have typescript declaration file?

### ISSUE THREAD

**Issue by andrewtw** [mikeal/r2#68]
That would be very helpful in typescript environment.

**Comment by ljharb** [comment]
Unless this project is written in TypeScript, the better place for definitions to go is usually DefinitelyTyped.

**Comment by andrewtw** [comment]
Yes, I do know that, but normally the definition file is gathered together by the community from project itself.

**Comment by Nebu1eto** [comment]
How about to try [this one](https://github.com/Hazealign/refetch)?
Disclaim: I made this library, and it written as TypeScript.

**Comment by phil-lgr** [comment]
given the widespread usage of TypeScript today and the size of this lib, it would be terrific to have it in TypeScript, node, http, node-fetch have excellent typings which we could rely on for r2
